### PR TITLE
Fix sales stock level to work off item reorder level instead of 0 or below #336

### PR DESCRIPTION
**Pull request by daN4cat**
Fix sales stock level to work off item reorder level instead of 0 or below #336

### CODE REVIEW

File: application/libraries/Sale_lib.php [hunk 1]
```diff
@@ -336,19 +336,19 @@ function add_item($item_id,$quantity=1,$item_location,$discount=0,$price=null,$d
...
 		if($this->validate_item($item_id) == false)
         {
             return false;
         }
-
-		
-		//$item = $this->CI->Item->get_info($item_id);
+		$item_info = $this->CI->Item->get_info($item_id);		
+		$data['reorder_level'] = $item_info->reorder_level;
 		$item_quantity = $this->CI->Item_quantity->get_item_quantity($item_id,$item_location)->quantity;
```
Comment (jekkos) [inline application/libraries/Sale_lib.php]: Why save this value in a associative array here? Can't we declare it directly?

Comment (odiea) [inline application/libraries/Sale_lib.php]: I'm not a programmer so that's what I came up with that worked. Please clean it up if you want to.

Comment (jekkos) [inline application/libraries/Sale_lib.php]: ok will do anytime soon

**Comment by jekkos** [comment]
I had a closer  look at this patch. Maybe we should show another warning then the quantity is insufficient? At least there are items left to sell so stock wise there's nothing really wrong as you're at least not having negative stock counts for the product at hand..

**Comment by odiea** [comment]
Yes that would be great. I just didn't want to play with the language files. The warning  will still work when it is 0 or below though.

**Comment by odiea** [comment]
Should 2 warnings be placed 1 for low inventory and 1 for 0 inventory ?

**Comment by jekkos** [comment]
Good point. Maybe just one or the other? Maybe the reorder level warning is only meaningful is the  level itself is higher than 0?

**Comment by odiea** [comment]
Yes. If it has both then someone can see that the level is low before going completely out.

**Comment by odiea** [comment]
Or change the wording to Your inventory level is low or out

**Comment by odiea** [comment]
So I duplicated the out_of_stock and set the new one to low inventory. I changed the warnings by duplicating them also. Does this sound like too much work or can it be set to work in the current low inventory function without duplicating these things.

**Comment by jekkos** [comment]
I've added a patch for this issue in latest master. An extra change in the code was needed to show the warnings when editing quantities directly.

**Pull request closed without merging**